jib-core/src/main/java/com/google/cloud/tools/jib/builder/steps/PullBaseImageStep.java

```diff
@@ -97,9 +98,19 @@ public ImagesAndRegistryClient call()
     // Skip this step if this is a scratch image
     ImageReference imageReference = buildContext.getBaseImageConfiguration().getImage();
     if (imageReference.isScratch()) {
+      ImmutableSet<Platform> platforms = buildContext.getContainerConfiguration().getPlatforms();
       eventHandlers.dispatch(LogEvent.progress("Getting scratch base image..."));
-      return new ImagesAndRegistryClient(
-          Collections.singletonList(Image.builder(buildContext.getTargetFormat()).build()), null);
+      if (platforms.isEmpty()) {
+        return new ImagesAndRegistryClient(
+            Collections.singletonList(Image.builder(buildContext.getTargetFormat()).build()), null);
+      }
```

Comment:
By the way we designed this, we can assume `platforms` is never empty. Let's just say
```java
Verify.verify(!platforms.isEmpty());
```